configuration.py

```diff
@@ -0,0 +1,153 @@
+                    input_date, "%Y-%m-%dT%H:%M:%S").timetuple()))
+            except Exception:
+                self.logger.warn(
+                    "The parameter {} is not in the right format. Expected format \"YYYY-MM-DDThh:mm:ss\", found: {}".format(param_name, input_date))
+                return False
+
+            if input_date > current_time:
+                self.logger.warn(
+                    "The parameter {} can not be greater than current Time".format(param_name))
+                return False
+
+            elif input_date < 0:
+                self.logger.warn(
+                    "The parameter {} should be greater than 01-Jan-1970 UTC".format(param_name))
+                return False
+        return True
+
+    def validate_interval(self, param_name, interval):
+        if not interval:
+            self.logger.warn(
+                "The parameter {} is not present. Considering the default value as 60 minutes".format(param_name))
+        elif not isinstance(interval, int):
+            self.logger.warn("The parameter {} should be a positive integer, found: {}. Considering the default value as 60 minutes".format(
+                param_name, interval))
+        elif not interval > 0:
+            self.logger.warn("The parameter {} must be a positive integer, found: {}. Considering the default value as 60 minutes".format(
+                param_name, interval))
+
+    def validate(self):
+        self.logger.info('Validating the configuration parameters')
+        # validating mandatory fields
+        if not self.configurations.get('sharepoint.client_id'):
+            print_and_log(self.logger, 'error',
+                          'sharepoint.clien_id cannot be empty')
+            return False
+        if not self.configurations.get('sharepoint.client_secret'):
+            print_and_log(self.logger, 'error',
+                          'sharepoint.client_secret cannot be empty')
+            return False
+        if not self.configurations.get('sharepoint.realm'):
+            print_and_log(self.logger, 'error',
+                          'sharepoint.realm cannot be empty')
+            return False
+        if not self.configurations.get('sharepoint.host_url'):
+            print_and_log(self.logger, 'error',
+                          'sharepoint.host_url cannot be empty')
+            return False
+        if not self.configurations.get('ws.access_token'):
+            print_and_log(self.logger, 'error',
+                          'ws.access_token cannot be empty')
+            return False
+        if not self.configurations.get('ws.source_id'):
+            print_and_log(self.logger, 'error', 'ws.source_id cannot be empty')
+            return False
+        if not self.configurations.get('ws.host_url'):
+            print_and_log(self.logger, 'error', 'ws.host_url cannot be empty')
+            return False
+        if not self.configurations.get('sharepoint.site_collections'):
```

Comment:
This seems unusual to require the Site Collections to be specified in configuration. Can we not list Site Collections via the API and thereby abstract this away for the user?